backend/app/Providers/AppServiceProvider.php

```
@@ -115,6 +115,26 @@ public function boot(): void
+        // Typed lifecycle events (previously defined but never fired) — these make
+        // WorkOrderCreated/Updated/Completed and StepStarted/Completed real on every
+        // save path; BatchCreated is fired via Batch::$dispatchesEvents.
+        \App\Models\WorkOrder::observe(\App\Observers\WorkOrderEventObserver::class);
+        \App\Models\BatchStep::observe(\App\Observers\BatchStepEventObserver::class);
+
+        // Generic CRUD hook: one wildcard Eloquent listener re-dispatches
+        // ResourceChanged for every curated resource (SoftDeleteRegistry::MODELS)
+        // so a module can hook any create/update/delete without per-model wiring.
+        $hookedModels = array_flip(array_values(\App\Support\SoftDeleteRegistry::MODELS));
+        foreach (['created', 'updated', 'deleted'] as $verb) {
+            Event::listen("eloquent.{$verb}: *", function (string $eventName, array $data) use ($hookedModels, $verb) {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$eventName'. (undefined)

(UnusedFormalParameter)